src/packages/core/icon-registry/icon.registry.ts

```diff
@@ -13,26 +12,68 @@ interface UmbIconDescriptor {
  * @description - Icon Registry. Provides icons from the icon manifest. Icons are loaded on demand. All icons are prefixed with 'icon-'
  */
 export class UmbIconRegistry extends UUIIconRegistry {
+	#initResolve?: () => void;
+	#init: Promise<void> = new Promise((resolve) => {
+		this.#initResolve = resolve;
+	});
+
+	#icons: UmbIconDescriptor[] = [];
+	#unhandledRequests: { name: string; provider: UUIIconHost }[] = [];
```

Comment:
Could this be a `Set` or do we need to handle each request individually for each icon host?

Reply:
I would say it could be a Map. So I've updated that :-)